petab/v2/petab1to2.py

```diff
@@ -152,3 +183,49 @@ def _copy_file(src: Path | str, dest: Path):
             return
     except FileNotFoundError:
         shutil.copy(str(src), str(dest))
+
+
+def _melt_condition_df(
+    condition_df: pd.DataFrame, model: v1.Model
+) -> pd.DataFrame:
+    """Melt condition table."""
+    condition_df = condition_df.copy().reset_index()
+    with suppress(KeyError):
+        # TODO: are condition names still supported in v2?
+        condition_df.drop(columns=[v2.C.CONDITION_NAME], inplace=True)
```

Comment:
Since I was so far effectively suggesting no condition names:
I can see that arbitrary metadata like descriptive names for a PEtab component is useful, if the ID itself isn't so interpretable. But this could exist in a separate metadata table with columns like e.g. `PETAB_ENTITY_ID`, `PETAB_ENTITY_NAME`, and perhaps even things like units, if people want to annotate. I have rarely/never used these though, but since one aim of this standard is reproducibility/interoperability, ideal there is sufficient metadata to ensure two different people interpret/use the problem similarly.

Reply:
Agreed that some extra metadata will often be useful. Format/location to be discussed. The yaml file would an alternative option, although less convenient if the metadata itself would be in a tabular format.